src/pwa/pages/_app.tsx

```diff
@@ -21,16 +25,72 @@ const lightTheme = createTheme(lightThemeOptions);
 
 const MyApp: React.FunctionComponent<MyAppProps> = (props) => {
   const { Component, emotionCache = clientSideEmotionCache, pageProps } = props;
+  const [loading, setLoading] = React.useState<boolean>(false);
+  const [globalLoading, setGlobalLoading] = React.useState<boolean>(true);
+
+
+  Router.events.on('routeChangeStart', (url) => {
+    setLoading(true);
+  });
+
+  Router.events.on('routeChangeComplete', (url) => {
+    setLoading(false);
+  });
+
+  useEffect(() => {
```

Comment:
Is there a way to wait for a meaningful event, e.g. contentLoaded, instead of showing loader for a fixed time?

Reply:
It is just 1s before showing the app. not a meaningful event or anything. If you don't like it. I will remove

Reply:
I have no problem with implementation, loader is needed to hide application while it loads content. I am just having concerns about duration for which loader is shown.
What if application takes 5 seconds to load content? Then its not enough time.
What if it takes 100ms to load? Then we are showing loader for 900ms more than needed.
If there is an event that would signal that content is ready, I would prefer to use it, instead of arbitrary time period.

Reply:
I provide 2 behaviors with loader:
1. the first one is a global loader. I set 1s duration for the first time the user opens the app. some styles are loaded after the DOM has rendered. So I'm investigating this problem. For now, I prevent it with 1s of loader.
2.  The second is the loader content of the page which is demanded in backlog, it will be shown when the user navigates between pages and hidden when the content of the page is ready. it is triggered by **routeChangeComplete** event.

Reply:
Sorry for not being specific. My issue was with global loader, which uses fixed timeout.

Reply:
I switch to using the dynamic import `Layout `component to fix the issue's `CSS `rendering. the content page is also applied loader